src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -218,11 +218,11 @@ static bool _validateInExpression(AR_ExpNode *exp) {
 	assert(exp->op.child_count == 2);
 
 	AR_ExpNode *list = exp->op.children[1];
-	// In case the array is a parameter such as "WHERE x IN $arr", evaluate the parameter for in-place replacement.
+	SIValue listValue = SI_NullVal();
+	bool constant_list = AR_EXP_ReduceToScalar(list, true, &listValue);
 	if(AR_EXP_IsParameter(list)) AR_EXP_Evaluate(list, NULL);
-	if(list->operand.type != AR_EXP_CONSTANT || list->operand.constant.type != T_ARRAY) return false;
+	if(!constant_list || SI_TYPE(listValue) != T_ARRAY) return false;
```

Comment:
I think checking if `SI_TYPE(listValue) != T_ARRAY` should be enough.

Reply:
I prefer this for my peace of mind.